### PR TITLE
Fixed Inconsistencies between Drive Paths on Windows (C:\ vs c:\)

### PR DESCRIPTION
**Pull request by johnrom**
It seems like VS Code or another dependency has begun using lowercase drive names in some places, but not in `${workspaceRoot}`

This will make it match up.

### CODE REVIEW

**Comment by felixfbecker** [comment]
Build is failing

**Comment by felixfbecker** [comment]
Referencing https://github.com/felixfbecker/vscode-php-debug/issues/239

**Comment by johnrom** [comment]
Prettier seems to be printing differently on my machine than in the build environment

**Comment by johnrom** [comment]
Builds fixed, ready for review

File: src/paths.ts [hunk 1]
```diff
@@ -53,7 +53,11 @@ export function convertClientPathToDebugger(localPath: string, pathMapping?: { [
     let localSourceRoot: string | undefined
     let serverSourceRoot: string | undefined
     let localFileUri = fileUrl(localPath, { resolve: false })
+    let localFileUriLower = fileUrl(localPath.replace(/[a-zA-Z]:\\/, (match: any) => match.toLowerCase()), {
```
Comment (felixfbecker) [inline src/paths.ts]: why is `match` `any`? Shouldn't this be inferred as `string`?

File: src/paths.ts [hunk 2]
```diff
@@ -75,12 +91,20 @@ export function convertClientPathToDebugger(localPath: string, pathMapping?: { [
             serverSourceRootUrl += '/'
         }
         // get the part of the path that is relative to the source root
-        const urlRelativeToSourceRoot = urlRelative(localSourceRootUrl, localFileUri)
+        const urlRelativeToSourceRoot = urlRelative(localSourceRootUrl, localFileUriLower)
         // resolve from the server source root
         serverFileUri = url.resolve(serverSourceRootUrl, urlRelativeToSourceRoot)
+
+        // we lowercased the Url, maybe we should undo that
```
Comment (felixfbecker) [inline src/paths.ts]: This comment confuses me, what do you mean with `maybe`?

File: src/paths.ts [hunk 3]
```diff
@@ -53,7 +53,11 @@ export function convertClientPathToDebugger(localPath: string, pathMapping?: { [
     let localSourceRoot: string | undefined
     let serverSourceRoot: string | undefined
     let localFileUri = fileUrl(localPath, { resolve: false })
+    let localFileUriLower = fileUrl(localPath.replace(/[a-zA-Z]:\\/, (match: any) => match.toLowerCase()), {
+        resolve: false,
+    })
     let serverFileUri: string
+    let uriWasLowerCased = false
```
Comment (felixfbecker) [inline src/paths.ts]: Frankly I am confused why this fix requires this variable - shouldn't it be possible to just _normalise_ everything to a canonical representation?

Comment (felixfbecker) [reply src/paths.ts]: E.g. before the path mapping feature I had this PR: https://github.com/felixfbecker/vscode-php-debug/pull/198
Can't the same be done, but for all paths in the map?

**Pull request merged**